dask/dataframe/rolling.py

```diff
@@ -0,0 +1,55 @@
+from functools import partial, wraps
+
+from toolz import merge
+import pandas as pd
+
+from .core import tokens
+
+
+def rolling_chunk(func, part1, part2, window, *args):
+    if part1.shape[0] < window:
+        raise NotImplementedError("Window larger than partition size")
+    if window > 1:
+        extra = window - 1
+        combined = pd.concat([part1.iloc[-extra:], part2])
+        applied = func(combined, window, *args)
+        return applied.iloc[extra:]
+    else:
+        return func(part2, window, *args)
+
+
+def wrap_rolling(func):
+    """Create a chunked version of a pandas.rolling_* function"""
+    @wraps(func)
+    def rolling(arg, window, *args, **kwargs):
+        if not isinstance(window, int):
+            raise TypeError('Window must be an integer')
+        if window < 0:
+            raise ValueError('Window must be a positive integer')
```

Comment:
pandas will actually catch this, but if dask is going to do it we should do it during dask construction, not during the computation